tools/rule_checker/main.go

```diff
@@ -19,29 +19,87 @@ package main
 import (
 	"flag"
 	"fmt"
-
-	"github.com/golang/glog"
+	"io"
+	"os"
 
 	"github.com/prometheus/prometheus/rules"
 )
 
-var ruleFile = flag.String("rule-file", "", "The path to the rule file to check.")
+var (
+	flagset  = flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	ruleFile = flagset.String("rule-file", "", "The path to the rule file to check. (Deprecated)")
+)
+
+// checkFile reads from in or the file if in is nil. Sucessfully read rules
+// are printed to out.
+func checkFile(filename string, in io.Reader, out io.Writer) error {
+	if in == nil {
+		f, err := os.Open(filename)
```

Comment:
How about assigning `in` here already instead of the extra variable and extra assignment below?